qml/components/TimeProgressBar.qml

```diff
@@ -31,30 +31,26 @@ Item {
     property int minValue: 0
     property int maxValue: 100
     property int value: 0
-
-    readonly property bool timeEditionInProgress: textEditableInput.visible
+    property bool timeEditMode: false
 
     implicitHeight: text.font.pixelSize*1.4
     implicitWidth: 200
 
     signal timeValueChanged(var newValue)
 
-    function startTimeEdition()
+    onTimeEditModeChanged:
     {
-        if(textEditableInput.visible)
+        if(timeEditMode)
         {
-            return
+            textEditableInput.visible = true
```

Comment:
Please see that `textEditableInput.visible` can be skipped here, as you already set a binding in `timeEditableInput`.
As for the focus I think it should be better to make it related to the `timeEditableInput` visibility not the edit mode itself.

What do you think?

Reply:
Done